src/Illuminate/Database/Query/Grammars/Grammar.php

```diff
@@ -504,9 +504,10 @@ protected function whereNested(Builder $query, $where)
         // Here we will calculate what portion of the string we need to remove. If this
         // is a join clause query, we need to remove the "on" portion of the SQL and
         // if it is a normal query we need to take the leading "where" of queries.
-        $offset = $query instanceof JoinClause ? 3 : 6;
+        $whereSql = $this->compileWheres($where['query']);
+        $offset = $query instanceof JoinClause  && str_starts_with($whereSql, 'on ') ? 3 : 6;
 
-        return '('.substr($this->compileWheres($where['query']), $offset).')';
+        return '('.substr($whereSql, $offset).')';
```

Comment:
this was actually a bug, nested inner where's would be cut off on 'whe' and still leave 're ' in the query